specs/Storage.Pickers/FileOpenPicker.md

```diff
@@ -47,6 +57,14 @@ var openPicker = new FileOpenPicker(this.AppWindow.Id)
     //     If not specified, the system uses a default label of "Open" (suitably translated).
     CommitButtonText = "Choose selected files",
 
+    // (Optional) group file types into labeled choices
+    //     FileTypeChoices takes precedence over FileTypeFilter when both defined.
+    FileTypeChoices = new Dictionary<string, IList<string>>
+    {
+        { "Documents", new List<string> { ".txt", ".doc", ".docx" } },
+        { "Pictures", new List<string> { ".png", ".jpg", ".jpeg", ".bmp" } }
```

Comment:
This design is tricky because the app is providing the IList, and it could provide one that behaves erratically. (E.g., reports the wrong Size, or one that generates different extensions each time you ask, or one that makes a cross-process COM call when you ask for an item.) We have to be careful how we access the list elements to be resilient to misbehaving list implementations.

```csharp
class WeirdList : IList<string>
{
    public int Count => Math.Random(0, 10);
    public bool IsReadOnly => true;
    public Item[int index] => GenerateRandomString();
}
```

Also, how do you set this in JavaScript or Python? I don't think they have a way to create IList implementations.

But it seems that this already exists in the FileSavePicker, so hopefully we can use the same solution.